### DISCUSSION TITLE
Connection disconnection depending on the domain visited

### DISCUSSION THREAD

**Discussion by Exhigh** [XTLS/Xray-core#1738] [Q&A]
So, i have tried configurations ranging from vless / vmess (grpc, ws, tcp, kcp) + tls and xtls, trojan (tls and non tls) and ss2022, but when using xray and when i try to connect U.BIsoft domain, i.e using the u.bisoft connect app, what happens is it looks for patches, logs in and then gets stuck at initializing and then fails. But when i use any of the above configuration but using routing, forward u.bi.com and u.bisoft.com domains to a wireguard configuration that is running within the same server the connection goes through and it works. Not really sure as to why this happens in particular. There also doesn't seem to be any real log errors that can be seen in the xray error logs while setting the xray log to "debug".

**Comment by Exhigh** [comment]
I could provide config files, but I don't really know if this is an isolated incident, as in whether it just affects me and not anyone else.

**Reply by Exhigh** [reply]
@RPRX

**Comment by ghost** [comment]
I have this problem too
It started this week
Using wireguard outbounds causes unstoppable RAM usage until it overflows and the service restarts
I tried many things but the problem was not solved
It used to be fine, maybe something changed about the 1.1.1.1 WARP service and this problem started
I didn't had access any other Wireguard services to check if it was related to WARP

Here's my config.json file

{"inbounds":[{"listen":"172.17.0.2","port":5443,"protocol":"vless","settings":{"clients":[{"id":"","level":0}],"decryption":"none"},"streamSettings":{"network":"ws","security":"tls","tlsSettings":{"alpn":["http/1.1"],"certificates":[{"certificateFile":"/root/cert/cert.pem","keyFile":"/root/cert/privkey.pem"}]}}}],"outbounds":[{"protocol":"wireguard","settings":{"secretKey":"","address":["172.16.0.2/32","2606:4700:110:82fe:3ce6:ee95:fbde:948c/128"],"peers":[{"publicKey":"bmXOC+F1FxEMF9dyiK2H5/1SUtzH0JuVo51h2wPfgyo=","endpoint":"162.159.192.1:500","keepAlive":15}],"mtu":1280,"workers":4}}],"dns":{"servers":["1.1.1.1"]},"log":{"access":"/var/log/xray/access.log","error":"/var/log/xray/error.log","loglevel":"error","dnsLog":true}}

**Reply by Exhigh** [reply]
> I have this problem too It started this week Using wireguard outbounds causes unstoppable RAM usage until it overflows and the service restarts I tried many things but the problem was not solved It used to be fine, maybe something changed about the 1.1.1.1 WARP service and this problem started I didn't had access any other Wireguard services to check if it was related to WARP
> 
> Here's my config.json file
> 
> {"inbounds":[{"listen":"172.17.0.2","port":5443,"protocol":"vless","settings":{"clients":[{"id":"","level":0}],"decryption":"none"},"streamSettings":{"network":"ws","security":"tls","tlsSettings":{"alpn":["http/1.1"],"certificates":[{"certificateFile":"/root/cert/cert.pem","keyFile":"/root/cert/privkey.pem"}]}}}],"outbounds":[{"protocol":"wireguard","settings":{"secretKey":"","address":["172.16.0.2/32","2606:4700:110:82fe:3ce6:ee95:fbde:948c/128"],"peers":[{"publicKey":"bmXOC+F1FxEMF9dyiK2H5/1SUtzH0JuVo51h2wPfgyo=","endpoint":"162.159.192.1:500","keepAlive":15}],"mtu":1280,"workers":4}}],"dns":{"servers":["1.1.1.1"]},"log":{"access":"/var/log/xray/access.log","error":"/var/log/xray/error.log","loglevel":"error","dnsLog":true}}

This isn't the issue that i am talking about, the wireguard service causing excessive RAM usage isnt' the issue that i brought up. The issue that i am facing is the issue that without the use of an outbound wireguard config for certain domains like u.bi.com and u.bisoft.com the connection doesn't go through

**Comment by ghost** [comment]
Ok then please provide the configuration so i check ubisoft app

**Reply by Exhigh** [reply]
Server: 

{
 "log": {
   "loglevel": "debug",
   "access": "/var/log/xray/access.log",
   "error": "/var/log/xray/error.log"
 },

 "inbounds": [
  {
    "listen": "127.0.0.1",
    "port": 52342,
    "protocol": "vless",
    "settings": {
        "clients": [
          {
            "email": "Chicken@xray.com",
            "id": "ab5436d2-34fb-4nm1-a276-890f91c24698",
            "level": 0
          }
         ],
         "decryption": "none"
    },
            "streamSettings": {
                "network": "ws",
                "wsSettings": {
                  "path": "/"
                },
                "mux": {
                 "enabled": true,
                 "concurrency": 8
                },
               "sockopt": {
                 "acceptProxyProtocol": false,
                 "tcpFastOpen": true,
                 "tcpcongestion": "bbr"
               }
            },
            "sniffing": {
                "enabled": true,
                "metadataonly": false,
                #"routeOnly": true,
                "destOverride": [
                 "http",
                 "tls",
                 "quic"
                ],
                "domainsExcluded": [
                  "rbsxbxp-mim.vivox.com",
                  "rbsxbxp.www.vivox.com",
                  "rbsxbxp-ws.vivox.com",
                  "rbspsxp.www.vivox.com",
                  "rbspsxp-mim.vivox.com",
                  "rbspsxp-ws.vivox.com",
                  "rbswxp.www.vivox.com",
                  "rbswxp-mim.vivox.com",
                  "disp-rbspsp-5-1.vivox.com",
                  "disp-rbsxbp-5-1.vivox.com",
                  "proxy.rbsxbp.vivox.com",
                  "proxy.rbspsp.vivox.com",
                  "proxy.rbswp.vivox.com",
                  "rbswp.vivox.com",
                  "rbsxbp.vivox.com",
                  "rbspsp.vivox.com",
                  "rbspsp.www.vivox.com",
                  "rbswp.www.vivox.com",
                  "rbsxbp.www.vivox.com",
                  "rbsxbxp.vivox.com",
                  "rbspsxp.vivox.com",
                  "rbswxp.vivox.com"
                 ]
           }
        }
    ],
    "outbounds": [
        {
            "protocol": "freedom",
            "settings": {
              "domainStrategy": "AsIs",
              "userLevel": 0
            },
            "tag": "direct"
        },
        {
            "protocol": "wireguard",
            "settings": {
              "secretKey": "#key",
              "address": [
                "172.34.0.3/32"
              ],
              "peers": [
               {
                "endpoint": "127.0.0.1:52341",
                "publicKey": "#key",
                "preSharedKey": "#key"
               }
              ]
            },
            "tag": "wg-1"
        },
        {
            "protocol": "wireguard",
            "settings": {
              "secretKey": "#key",
              "address": [
                "172.34.0.5/32"
              ],
              "peers": [
               {
                "endpoint": "127.0.0.1:52341",
                "publicKey": "#key",
                "preSharedKey": "#key"
               }
              ]
            },
            "tag": "wg-2"
        },
        {
          "tag": "DNS-Internal",
          "protocol": "dns",
            "settings": {
               "address": "127.0.0.53",
               "port": 53
            }
        }
    ],
   "routing": {
     "domainStrategy": "AsIs",
        "rules": [
         {
          "type": "field",
          "port": 53,
          "network": "tcp,udp",
          "outboundTag": "DNS-Internal"
         },
          {
           "type": "field",
           "network": "tcp,udp",
           "user": ["ChickenX@temp.com"],
           "domain": ["rockstargames.com","ubisoft.com", "ubi.com"],
           "outboundTag": "wg-2"
         },
         {
           "type": "field",
           "network": "tcp,udp",
           "domain": ["rockstargames.com","ubisoft.com", "ubi.com"],
           "outboundTag": "wg-1"
         }
        ]
   }
}


Client : 


{
  "inbounds": [
    {
      "port": 2801,
      "protocol": "socks",
      "listen": "127.0.0.1",
      "settings": {
        "auth": "noauth",
        "udp": true
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "vless",
      "settings": {
        "vnext": [
          {
            "address": "#address",
            "port": 443,
            "users": [
              {
                "id": "ab5436d2-34fb-4nm1-a276-890f91c24698",
                "alterId": 0,
                "encryption": "none",
                "flow": ""
              }
            ]
          }
        ],
        "packetEncoding": "xudp"
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "tlsSettings": {
          "allowInsecure": false,
          "serverName": "#address",
          "fingerprint": "random"
        },
        "wsSettings": {
          "path": "/#path?ed=2048",
          "headers": {
            "Host": "#address"
          }
        },
        "sockopt": {
          "tcpFastOpen": true
        }
      },
      "mux": {
        "enabled": true,
        "packetEncoding": "xudp",
        "concurrency": 8
      }
    }
  ]
}

**Reply by Exhigh** [reply]
so this is the configuration by which the connection to the above mentioned domains you will need to remove the wireguard routing in the server to replicate the issue i was mentioning
